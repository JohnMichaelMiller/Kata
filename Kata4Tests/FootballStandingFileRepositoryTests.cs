using Microsoft.VisualStudio.TestTools.UnitTesting;
using Kata4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Kata4.Tests
{
    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class FootballStandingFileRepositoryTests
    {
        [TestMethod()]
        public void FootballStandingFileRepositoryTest()
        {
            var repo = new FootballStandingFileRepository();
            Approvals.VerifyAll(repo, "Football Standing");
        }
    [TestMethod()]
    public void FootballStandingFileRepositoryHasCorrectMinimumTemperatureDifferential()
    {
        var repo = new FootballStandingFileRepository();
        var result =
            from w in repo
            where -w.GoalDifferential ==
                repo.Min(d => -d.GoalDifferential)
            select w;

        Approvals.VerifyAll(result, "Football Standing");
    }

    }
}