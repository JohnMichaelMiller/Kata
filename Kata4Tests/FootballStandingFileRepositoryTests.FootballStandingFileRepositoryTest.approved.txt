Football Standing[0] = Rank:1 Team: Arsenal, Goals For 79, Goals Against: 36, Goal Diff: 43
Football Standing[1] = Rank:2 Team: Liverpool, Goals For 67, Goals Against: 30, Goal Diff: 37
Football Standing[2] = Rank:3 Team: Manchester_U, Goals For 87, Goals Against: 45, Goal Diff: 42
Football Standing[3] = Rank:4 Team: Newcastle, Goals For 74, Goals Against: 52, Goal Diff: 22
Football Standing[4] = Rank:5 Team: Leeds, Goals For 53, Goals Against: 37, Goal Diff: 16
Football Standing[5] = Rank:6 Team: Chelsea, Goals For 66, Goals Against: 38, Goal Diff: 28
Football Standing[6] = Rank:7 Team: West_Ham, Goals For 48, Goals Against: 57, Goal Diff: -9
Football Standing[7] = Rank:8 Team: Aston_Villa, Goals For 46, Goals Against: 47, Goal Diff: -1
Football Standing[8] = Rank:9 Team: Tottenham, Goals For 49, Goals Against: 53, Goal Diff: -4
Football Standing[9] = Rank:10 Team: Blackburn, Goals For 55, Goals Against: 51, Goal Diff: 4
Football Standing[10] = Rank:11 Team: Southampton, Goals For 46, Goals Against: 54, Goal Diff: -8
Football Standing[11] = Rank:12 Team: Middlesbrough, Goals For 35, Goals Against: 47, Goal Diff: -12
Football Standing[12] = Rank:13 Team: Fulham, Goals For 36, Goals Against: 44, Goal Diff: -8
Football Standing[13] = Rank:14 Team: Charlton, Goals For 38, Goals Against: 49, Goal Diff: -11
Football Standing[14] = Rank:15 Team: Everton, Goals For 45, Goals Against: 57, Goal Diff: -12
Football Standing[15] = Rank:16 Team: Bolton, Goals For 44, Goals Against: 62, Goal Diff: -18
Football Standing[16] = Rank:17 Team: Sunderland, Goals For 29, Goals Against: 51, Goal Diff: -22
Football Standing[17] = Rank:18 Team: Ipswich, Goals For 41, Goals Against: 64, Goal Diff: -23
Football Standing[18] = Rank:19 Team: Derby, Goals For 33, Goals Against: 63, Goal Diff: -30
Football Standing[19] = Rank:20 Team: Leicester, Goals For 30, Goals Against: 64, Goal Diff: -34
